api/trace/api.go

```diff
@@ -58,17 +58,30 @@ type (
 		Tracer() Tracer
 
 		Finish()
+
+		// IsRecordingEvents returns true is the span is active and recording events is enabled.
+		IsRecordingEvents() bool
+
+		// SpancContext returns span context of the span. Return SpanContext is usable
+		// even after the span is finished.
+		SpanContext() core.SpanContext
+
+		SetStatus(core.Status)
 	}
 
 	Injector interface {
 		Inject(core.SpanContext, tag.Map)
 	}
 
-	Option struct {
-		attribute  core.KeyValue
-		attributes []core.KeyValue
-		startTime  time.Time
-		reference  Reference
+	// Option apply changes to SpanOptions.
+    Option func(*SpanOptions)
```

Comment:
nit: bad spacing

May also want to consider a different name. This will end up being `trace.Option`, but maybe it should be `trace.SpanOption` ?

Reply:
SpanOption would then be confusing with SpanOptions.

Reply:
What if SpanOptions was renamed, not exported, or combined with a Span type?

Reply:
How about 
```go
SpanOption func(*SpanOptions) . // If SpanOptions is exported.
// OR
SpanOption func(*spanOptions) . // If SpanOptions is not exported.

```
I see this pattern in general.

Reply:
@freeformz I have made the renamed Option to SpanOption. I'll merge this for now and open another issue if it should be renamed.